app/assets/javascripts/views/courses.js

```diff
@@ -101,5 +111,27 @@ Yacs.views.courses = function (data) {
       c.classList[isSelected ? 'remove' : 'add']('selected');
     });
     if (isCourseSelected(c)) c.classList.add('selected');
+
+    var desc = c.querySelector('course-description');
+    var showButton = c.querySelector('show-hide-button');
+    Yacs.on('click', showButton, function (showButton, event) {
+      var descTruncated = desc.classList.contains("truncated");
+      desc.classList[descTruncated ? 'remove' : 'add']("truncated");
+      showButton.classList[descTruncated ? 'add' : 'remove']("open");
+      event.stopPropagation();
+    });
+    requireTruncation(desc, showButton);
+  });
+
+  var resizingFxn;
+  window.addEventListener("resize", function () {
```

Comment:
Yeah, you just pointed out a major problem I overlooked before with binding the event to the window. I could just remove the resize check for now. Truncating description and the show/hide button work find without it.